feg/gateway/services/n7_n40_proxy/n7/config.go

```diff
@@ -0,0 +1,132 @@
+	"magma/feg/gateway/utils"
+	"magma/gateway/mconfig"
+)
+
+const (
+	N7N40ProxyServiceName = "n7_n40_proxy"
+
+	PcfApiRoot            = "PCF_API_ROOT"
+	PcfTokenUrl           = "PCF_TOKEN_URL"
+	PcfClientId           = "PCF_CLIENT_ID"
+	PcfClientSecret       = "PCF_CLIENT_SECRET"
+	N7ClientLocalAddr     = "N7_CONSUMER_LOCAL_ADDR"
+	N7ClientNotifyApiRoot = "N&_CONSUMER_NOTIFY_API_ROOT"
+
+	DefaultPcfApiRoot      = "https://localhost"
+	DefaultPcfTokenUrl     = "https://localhost/token"
+	DefaultClientId        = "magma_client_id"
+	DefaultClientSecret    = "magma_client_secret"
+	DefaultN7ClientAddr    = "localhost:0"
+	DefaultN7ClientApiRoot = "https://localhost/npcf-smpolicycontrol/v1"
+)
+
+type PCFConfig struct {
+	ApiRoot      url.URL
+	TokenUrl     string
+	ClientId     string
+	ClientSecret string
+}
+
+type N7ClientConfig struct {
+	LocalAddr     string // ip:port or :port
+	NotifyApiRoot string
+}
+
+type N7Config struct {
+	DisableN7 bool
+	Server    PCFConfig
+	Client    N7ClientConfig
+}
+
+func GetN7Config() (*N7Config, error) {
+	configPtr := &mcfgprotos.N7N40ProxyConfig{}
+	conf := &N7Config{}
+
+	err := mconfig.GetServiceConfigs(N7N40ProxyServiceName, configPtr)
+	if err != nil || !validManagedConfig(configPtr) {
+		glog.V(2).Infof("Managed Configs Load Error: %v Using EnvVars", err)
+		apiRoot, err := url.ParseRequestURI(utils.GetValueOrEnv("", PcfApiRoot, DefaultPcfApiRoot))
+		if err != nil {
+			return nil, fmt.Errorf("invalid Server ApiRoot - %s", err)
+		}
+		conf.Server = PCFConfig{
+			ApiRoot:      *apiRoot,
+			TokenUrl:     utils.GetValueOrEnv("", PcfTokenUrl, DefaultPcfTokenUrl),
+			ClientId:     utils.GetValueOrEnv("", PcfClientId, DefaultClientId),
+			ClientSecret: utils.GetValueOrEnv("", PcfClientSecret, DefaultClientSecret),
+		}
+		conf.DisableN7 = false
+		conf.Client = N7ClientConfig{
+			LocalAddr:     utils.GetValueOrEnv("", N7ClientLocalAddr, DefaultN7ClientAddr),
+			NotifyApiRoot: utils.GetValueOrEnv("", N7ClientNotifyApiRoot, DefaultN7ClientApiRoot),
+		}
+	} else {
+		n7configPtr := configPtr.N7Config
+		conf.DisableN7 = n7configPtr.DisableN7
+		apiRoot, err := url.ParseRequestURI(utils.GetValueOrEnv("", PcfApiRoot, n7configPtr.Server.GetApiRoot()))
+		if err != nil {
+			return nil, fmt.Errorf("invalid Server ApiRoot - %s", err)
+		}
+		conf.Server = PCFConfig{
+			ApiRoot:      *apiRoot,
+			TokenUrl:     utils.GetValueOrEnv("", PcfTokenUrl, n7configPtr.Server.GetTokenUrl()),
+			ClientId:     utils.GetValueOrEnv("", PcfClientId, n7configPtr.Server.GetClientId()),
+			ClientSecret: utils.GetValueOrEnv("", PcfClientSecret, n7configPtr.Server.GetClientSecret()),
+		}
+		conf.Client = N7ClientConfig{
+			LocalAddr:     utils.GetValueOrEnv("", N7ClientLocalAddr, n7configPtr.Client.LocalAddr),
+			NotifyApiRoot: utils.GetValueOrEnv("", N7ClientNotifyApiRoot, n7configPtr.Client.NotifyApiRoot),
+		}
+	}
+	err = validateN7Config(conf)
+	if err != nil {
+		return nil, err
+	}
+	return conf, nil
```

Comment:
you don't need to change it but in those cases sometimes we do

```
return validateN7Config(conf)
```

where validateN7Config is
```
func validateN7Config(config *N7Config) (*N7Config, error) {
```